pennylane/ops/op_math/controlled_ops.py

```diff
@@ -334,3 +340,894 @@ def compute_matrix():  # pylint: disable=arguments-differ
+
+        .. seealso:: :meth:`~.CRX.matrix`
+
+        Args:
+            theta (tensor_like or float): rotation angle
+
+        Returns:
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> qml.CRX.compute_matrix(torch.tensor(0.5))
+        tensor([[1.0+0.0j, 0.0+0.0j,    0.0+0.0j,    0.0+0.0j],
+                [0.0+0.0j, 1.0+0.0j,    0.0+0.0j,    0.0+0.0j],
+                [0.0+0.0j, 0.0+0.0j, 0.9689+0.0j, 0.0-0.2474j],
+                [0.0+0.0j, 0.0+0.0j, 0.0-0.2474j, 0.9689+0.0j]])
+        """
+        interface = qml.math.get_interface(theta)
+
+        c = qml.math.cos(theta / 2)
+        s = qml.math.sin(theta / 2)
+
+        if interface == "tensorflow":
+            c = qml.math.cast_like(c, 1j)
+            s = qml.math.cast_like(s, 1j)
+
+        # The following avoids casting an imaginary quantity to reals when back propagating
+        c = (1 + 0j) * c
+        js = -1j * s
+        ones = qml.math.ones_like(js)
+        zeros = qml.math.zeros_like(js)
+        matrix = [
+            [ones, zeros, zeros, zeros],
+            [zeros, ones, zeros, zeros],
+            [zeros, zeros, c, js],
+            [zeros, zeros, js, c],
+        ]
+
+        return qml.math.stack([stack_last(row) for row in matrix], axis=-2)
+
+    @staticmethod
+    def compute_decomposition(phi, wires):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators (static method). :
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+
+        .. seealso:: :meth:`~.CRot.decomposition`.
+
+        Args:
+            phi (float): rotation angle :math:`\phi`
+            wires (Iterable, Wires): the wires the operation acts on
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> qml.CRX.compute_decomposition(1.2, wires=(0,1))
+        [RZ(1.5707963267948966, wires=[1]),
+        RY(0.6, wires=[1]),
+        CNOT(wires=[0, 1]),
+        RY(-0.6, wires=[1]),
+        CNOT(wires=[0, 1]),
+        RZ(-1.5707963267948966, wires=[1])]
+
+        """
+        pi_half = qml.math.ones_like(phi) * (np.pi / 2)
+        return [
+            qml.RZ(pi_half, wires=wires[1]),
+            qml.RY(phi / 2, wires=wires[1]),
+            qml.CNOT(wires=wires),
+            qml.RY(-phi / 2, wires=wires[1]),
+            qml.CNOT(wires=wires),
+            qml.RZ(-pi_half, wires=wires[1]),
+        ]
+
+    def decomposition(self):
+        r"""Representation of the operator as a product of other operators.
+
+        .. math:: O = O_1 O_2 \dots O_n
+
+        A ``DecompositionUndefinedError`` is raised if no representation by decomposition is defined.
+
+        .. seealso:: :meth:`~.Operator.compute_decomposition`.
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+        return self.compute_decomposition(*self.parameters, wires=self.wires)
```

Comment:
So if we delete these lines, we will actually still get the exact same behaviour for the decomposition.

I think we should be able to just delete all these lines and be perfectly fine.

Reply:
Sounds good I'll do it.